### PR TITLE
Create ec_si.py

### PR DESCRIPTION
**Pull request by DanielNwaeze**
Added fiboa template for Slovenia Eurocrops dataset for PR

### CODE REVIEW

File: fiboa_cli/datasets/ec_si.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+
+from ..util import download_file, log
+from ..convert_utils import convert as convert_
+
+# File to read the data from
+# Can read any tabular data format that GeoPandas can read through read_file()
+# Supported protcols: HTTP(S), GCS, S3, or the local file system
+URI = "https://zenodo.org/records/8229128/files/SI_2021.zip?download=1"
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:12]: Any reason why this is running on EuroCrops v9 and not v10?

Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:12]:
```suggestion
URI = "https://zenodo.org/records/10118572/files/SI_2021.zip?download=1"
```

Any specific reason why this is running on EuroCrops v9 and not v10?

File: fiboa_cli/datasets/ec_si.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+- `EC_trans_n`: The original crop name translated into English
+- `EC_hcat_n`: The machine-readable HCAT name of the crop
+- `EC_hcat_c`: The 10-digit HCAT code indicating the hierarchy of the crop
+"""
+# Bounding box of the data in WGS84 coordinates
+BBOX = [13.410659993139568, 45.42227046875208, 16.596303417885895, 46.87424125552635]
+
+# Provider name, can be None if not applicable, must be provided if PROVIDER_URL is provided
+PROVIDER_NAME = "Eurocrops"
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:39]:
```suggestion
PROVIDER_NAME = "EuroCrops"
```

File: fiboa_cli/datasets/ec_si.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+# File to read the data from
+# Can read any tabular data format that GeoPandas can read through read_file()
+# Supported protcols: HTTP(S), GCS, S3, or the local file system
+URI = "https://zenodo.org/records/8229128/files/SI_2021.zip?download=1"
+FILENAME = "SI_2021_EC21.shp"
+
+# Unique identifier for the collection
+ID = "ec_si"
+# Title of the collection
+TITLE = "Field boundaries for Slovenia from Eurocrops (2021)"
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:18]:
```suggestion
TITLE = "Field boundaries for Slovenia from EuroCrops (2021)"
```

File: fiboa_cli/datasets/ec_si.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+# Description of the collection. Can be multiline and include CommonMark.
+DESCRIPTION = """
+This dataset contains the field boundaries for all of Slovenia in 2021. The data was collected by the Slovenian government and harmonized
+by the Eurocrops project and is available on Zenodo.
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:22]:
```suggestion
by the EuroCrops project and is available on Zenodo.
```

File: fiboa_cli/datasets/ec_si.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+            'type': 'string'
+        },
+        'EC_trans_n': {
+            'type': 'string'
+        },
+        'EC_hcat_n': {
+            'type': 'string'
+        },
+        'EC_hcat_c': {
+            'type': 'uint32'
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:121]: I'm getting:

> EC_hcat_c: Can't convert to uint32: cannot convert float NaN to integer
> ("Could not convert '3301010600' with type str: tried to convert to uint32", 'Conversion failed for column EC_hcat_c with type object')

**Review by m-mohr — CHANGES_REQUESTED** [review]

File: fiboa_cli/datasets/ec_si.py (hunk 1 above)
Comment (DanielNwaeze) [reply fiboa_cli/datasets/ec_si.py:12]: I used the link Chris provided in: https://github.com/fiboa/data/issues/31#issue-2288114517

File: fiboa_cli/datasets/ec_si.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+# Custom function to migrate the GeoDataFrame if the other options are not sufficient
+# This should be the last resort!
+# Function signature:
+#   func(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame
+MIGRATION = None
+
+# Schemas for the fields that are not defined in fiboa
+# Keys must be the values from the COLUMNS dict, not the keys
+MISSING_SCHEMAS = {
+    'required': ['gerk_pid', 'crop_type_class', 'rastlina', 'crop_latin', 'color', 'EC_trans_n','EC_hcat_n', 'EC_hcat_c'], # i.e. non-nullable properties
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:97]:
```suggestion
    'required': ['gerk_pid', 'crop_type_class', 'rastlina', 'crop_latin', 'color', 'EC_trans_n','EC_hcat_n'], # i.e. non-nullable properties
```

to fix https://github.com/fiboa/cli/pull/45/files#r1634625809

File: fiboa_cli/datasets/ec_si.py (hunk 1 above)
Comment (m-mohr) [reply fiboa_cli/datasets/ec_si.py:12]: I guess he just copied them randomly, let's go for the latest version.

File: fiboa_cli/datasets/ec_si.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            'type': 'string'
+        },
+        'EC_trans_n': {
+            'type': 'string'
+        },
+        'EC_hcat_n': {
+            'type': 'string'
+        },
+        'EC_hcat_c': {
+            'type': 'string'
```
Comment (m-mohr) [inline fiboa_cli/datasets/ec_si.py:121]: The change to string was not the intention. It's a numerical value, I think, but it was just lacking the nullability as the column was required, see my other comments.

Comment (m-mohr) [reply fiboa_cli/datasets/ec_si.py]: Now the schema for EC_hcat_c is fully gone. The uint32 in the beginning was fine, you just had to make it optional (as it now is).

**Pull request merged**